### PR TITLE
SW-5663 Automatically create project with the same name after an organization is created, and go to application

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2904?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2904** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2904?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2902** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2902?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2898** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2898?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2897** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2897?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2895** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2895?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/emptyStatePages/NoOrgApplicationLandingPage.tsx [hunk 1]
```diff
@@ -16,14 +21,39 @@ const EMPTY_STATE_CONTENT_STYLES = {
...
+        return;
+      }
+
+      // Create application with new project
+      const applicationResult = await ApplicationService.createApplication(projectResult.data.id);
+      if (!(applicationResult.data && applicationResult.requestSucceeded)) {
+        toastError(strings.ERROR_CREATE_APPLICATION);
+        goToApplicationList();
+        return;
+      }
```
Comment (nickgraz) [inline src/components/emptyStatePages/NoOrgApplicationLandingPage.tsx:51]: This seems like another set of requests that should be handled in a single service.

File: src/components/OrganizationsDropdown.tsx [hunk 2]
```diff
@@ -43,9 +47,24 @@ export default function OrganizationsDropdown(): JSX.Element {
     }
   };
 
+  const onOrgCreated = useCallback(
```
Comment (nickgraz) [inline src/components/OrganizationsDropdown.tsx:50]: Can you add this to the `OrganizationProvider`? Maybe you can just add a prop to `saveOrganization` like `redirectAndNotify`

File: src/components/emptyStatePages/NoOrgApplicationLandingPage.tsx (hunk 1 above)
Comment (tommylau523) [reply src/components/emptyStatePages/NoOrgApplicationLandingPage.tsx:51]: Added compound requests and redux in this PR!
https://github.com/terraware/terraware-web/pull/2913

Will rebase and apply new calls in this PR.

File: src/components/OrganizationsDropdown.tsx (hunk 2 above)
Comment (tommylau523) [reply src/components/OrganizationsDropdown.tsx:50]: Happy to. There are actually a few callback patterns I would like to use for ApplicationProvider too. So be on the look out for this PR or next!

**Review by nickgraz — APPROVED** [review]
Looks good!

**Comment by tommylau523** [comment]
### Merge activity

* **Jul 29, 5:11 PM EDT**: @tommylau523 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2904).

**Pull request merged**